### PR TITLE
Angular cleanup phase0

### PR DESCRIPTION
**Pull request by davidshq**
## Description

Phase 0 of the AngularJS cleanup: establish modern JS tooling for the importmap/Stimulus stack, introduce shared HTTP/navigation helpers, wire them into existing Stimulus controllers, and remove one dead Angular event-bus broadcast.

### Modern JS tooling (`app/javascript/`)

- **Vitest** for unit tests (`vitest.config.js`, `yarn test:unit`, Stimulus stub in `app/javascript/test/stimulus_stub.js`)
- **ESLint** for the full modern tree (`eslint.config.mjs`, scoped via `config/javascript_lint_scope.mjs`)
- **Prettier** for `api/` and `utils/` only (avoids mass-reformatting controllers/modules in this PR)
- Staged lint scripts (`bin/eslint-staged`, `bin/prettier-staged`), pre-commit hooks, CircleCI step, and `rake test:frontend` integration
- New **`docs/js_tooling.md`** documenting the split Angular vs modern JS worlds and how to run each tool

### Shared Stimulus helpers

- **`api/fetch`** — centralized `fetch` wrapper with automatic CSRF injection; supports header objects, tuple arrays, and `Headers` instances
- **`utils/quepid_root`** — reads `data-quepid-root-url` from `<body>` for client-side redirects when the server cannot pass a URL
- **`quepid_root_url`** Rails helper exposed on all layouts (`admin`, `application`, `core`) for subpath-aware root URLs
- Stimulus controllers migrated to `apiFetch`: `import_case`, `bulk_judgement`, `mapper_wizard`

### Angular cleanup

- Remove dead `updatedScorersList` event-bus broadcast from `scorerSvc` (no subscribers; `ScorerCtrl` loads via `scorerSvc.list()` directly)
- Fix `caseSvc.archiveCase` to use `findIndex` and skip `splice` when the case is not in `allCases`
- Update `event_bus_inventory.md` to reflect the removed broadcast

### Tests

- Vitest: `api/fetch.test.js`, `import_case_controller.test.js`, `quepid_root.test.js`
- Karma: `caseSvc_spec.js` coverage for `archiveCase` edge case
- Rails: `cases_controller_test.rb`, `application_helper_test.rb` (subpath `quepid_root_url`)
- Playwright: `stimulus_pages.spec.ts` smoke tests for Stimulus-backed pages

## Motivation and Context

Quepid runs two parallel JavaScript stacks during the Angular → Hotwire migration. Legacy Angular assets stay on JSHint/Karma; the importmap/Stimulus tree had no linting or fast unit-test infrastructure. This PR lays that foundation and starts standardizing HTTP/navigation patterns before migrating more controllers off Angular.

Builds on the parent `angular-cleanup` branch (dead Angular case UI removal + removal inventory docs).

## How Has This Been Tested?

- `bin/docker r yarn test:unit` (Vitest)
- `bin/docker r yarn test` (Karma — `caseSvc` specs)
- `bin/docker r rails test test/controllers/cases_controller_test.rb test/helpers/application_helper_test.rb`
- `bin/docker r yarn lint:js` / `bin/docker r yarn format:js:check`
- `bin/docker r yarn test:e2e test/playwright/stimulus_pages.spec.ts` (with app running)

## Types of changes

- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/javascript/api/fetch.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  })
+})
+
+describe("apiFetch", () => {
+  beforeEach(() => {
+    document.head.innerHTML = '<meta name="csrf-token" content="page-token">'
+    vi.stubGlobal(
+      "fetch",
+      vi.fn(() => Promise.resolve(new Response("ok")))
+    )
```
[the remark this answers is not in the document]
Comment (davidshq) [reply app/javascript/api/fetch.test.js:25]: The Headers test now uses the same case-insensitive key lookup as the CSRF Headers test below it.
